src/packages/dynamic-root/modals/dynamic-root-query-step-picker-modal.element.ts

```diff
@@ -0,0 +1,106 @@
+import { UmbDocumentTypePickerContext } from '../../documents/document-types/components/input-document-type/input-document-type.context.js';
+import { UmbId } from '@umbraco-cms/backoffice/id';
+import { UmbModalBaseElement } from '@umbraco-cms/backoffice/modal';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { css, html, customElement, map, state, ifDefined } from '@umbraco-cms/backoffice/external/lit';
+import { umbExtensionsRegistry } from '@umbraco-cms/backoffice/extension-registry';
+import type { UmbTreePickerDynamicRootQueryStep } from '@umbraco-cms/backoffice/components';
+import type { ManifestDynamicRootQueryStep } from '@umbraco-cms/backoffice/extension-registry';
+
+@customElement('umb-dynamic-root-query-step-picker-modal')
+export class UmbDynamicRootQueryStepPickerModalModalElement extends UmbModalBaseElement {
+	@state()
+	private _querySteps: Array<ManifestDynamicRootQueryStep> = [];
+
+	#documentTypePickerContext = new UmbDocumentTypePickerContext(this);
+
+	constructor() {
+		super();
+
+		this.observe(
+			umbExtensionsRegistry.byType('dynamicRootQueryStep'),
+			(querySteps: Array<ManifestDynamicRootQueryStep>) => {
+				this._querySteps = querySteps;
+			},
+		);
+	}
+
+	#choose(item: ManifestDynamicRootQueryStep) {
+		this.#openDocumentTypePicker(item.meta.queryStepAlias);
+	}
+
+	#close() {
+		this.modalContext?.reject();
+	}
+
+	#openDocumentTypePicker(alias: string) {
+		this.#documentTypePickerContext
+			.openPicker({
+				hideTreeRoot: true,
+				pickableFilter: (x) => x.isElement === false,
+			})
+			.then(() => {
+				const selectedItems = this.#documentTypePickerContext.getSelection();
+				this.#submit({
+					unique: UmbId.new(),
+					alias: alias,
+					anyOfDocTypeKeys: selectedItems,
+				});
+			});
+	}
```

Comment:
Consider the async/await pattern here as well

Reply:
Implemented! 👍